final-project/static/scripts.js

```diff
@@ -6,74 +6,93 @@
 // Enforce strict js code conventions
 "use strict";
 
-// Render the product on screen.
-$(function retrieveJSON() {
-
-  // Get the element id attribute when a product is clicked
-  $("#recently_added_products").click(function(event) {
-    // Get id of clicked element's parent
-    var product_id = $(event.target).parent().attr("id");
-    console.log("element id: ")
+// Keep track of the products we have on screen.
+var productWasClicked = document.querySelector(".productListView")
+
+
+// Search for product information when the user selects a product.
+productWasClicked.onclick = function() {
+
+  // Get the product information our user wants.
+  retrieveJSON( getIdOnClick(productWasClicked) );
+
+}
+
+
+// Retrieve product information in the form of a JSON.
+function retrieveJSON(product_id) {
+
+    console.log(" ")
+    console.log("retreiveJSON function started")
+    console.log("product_id = ")
     console.log(product_id)
 
+  // Create parameters for Flask.url_for() method.
+  var parameters = {
+    id: product_id
+  }
+  console.log(" ")
+  console.log("retrieveJSON returned parameters:")
+  console.log(parameters)
 
-    // Create parameters for Flask.url_for() method.
-    var parameters = {
-      id: product_id
-    }
-    console.log("parameters:")
-    console.log(parameters)
+  // Get the JSON with $.getJSON() & Flask.url_for().
+  // Use Flask.url_for("product", parameters) to generate JSON url.
+  var jsPromise = Promise.resolve(
```

Comment:
Why name this `jsPromise`? Why not give it a name that matches more of what its doing business-wise?

`getProductJsonUrl` or something

Reply:
Done.

But, it's in the latest PR.